services/clojars/clojars-base.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const Joi = require('joi')
+const { version: versionColor } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService, InvalidResponse } = require('..')
+
+const clojarsSchema = Joi.object({
+  downloads: nonNegativeInteger,
+  latest_version: Joi.string().required(),
+  latest_release: Joi.string().required(),
+}).required()
+
+class BaseClojarsService extends BaseJsonService {
+  async fetch({ clojar }) {
+    const url = `https://clojars.org/api/artifacts/${clojar}`
+    return this._requestJson({
+      url,
+      schema: clojarsSchema,
+      errorMessages: {
+        404: 'not found',
+      },
+    })
+  }
+}
+
+class ClojarsVersionService extends BaseClojarsService {
+  static render({ clojar, version }) {
+    return {
+      message: `[${clojar} "${version}"]`,
+      color: versionColor(version),
+    }
+  }
+
+  static transform(json, version) {
+    if (version === 'release') {
+      return { version: json.latest_release }
+    } else if (version === 'version') {
+      return { version: json.latest_version }
+    } else {
+      throw new InvalidResponse({
+        prettyMessage: new Error('version is invalid or null'),
```

Comment:
you can pass your desired message as a string (not an error) via `prettyMessage`. Please change to:

```suggestion
        prettyMessage: 'version is invalid or null',
```

Reply:
converted `prettyMessage` to string